### PR TITLE
PLNSRVCE-758: Detect HACBS workflow based on apibinding

### PR DESCRIPTION
**Pull request by Michkov**
Check if 'integration-service' apibinding exists if so then set HACBS workflow.

### What does this PR do?:
Detects HACBS workflow based on APIBinding.

### Which issue(s)/story(ies) does this PR fixes:
[PLNSRVCE-758](https://issues.redhat.com//browse/PLNSRVCE-758)

### PR acceptance criteria:

- [ ] Unit/Functional tests

  

- [ ] Documentation 

   

- [ ] Client Impact

### CODE REVIEW

File: gitops/prepare/prepare.go [hunk 1]
```diff
@@ -110,8 +114,13 @@ func ResolveBuildBundle(ctx context.Context, localClient, cli client.Client, nam
 
 // Return true when hacbs configmap exists in the namespace
```
Comment (psturc) [inline gitops/prepare/prepare.go:115]: nit: might be worth to update the description of the function

File: gitops/prepare/prepare.go [hunk 2]
```diff
@@ -110,8 +114,13 @@ func ResolveBuildBundle(ctx context.Context, localClient, cli client.Client, nam
...
 func IsHACBS(ctx context.Context, cli client.Client, namespace string) bool {
+	var apiBinding = kcpv1alpha1.APIBinding{}
+	err := cli.Get(ctx, types.NamespacedName{Name: HACBSAPIBindingName}, &apiBinding)
+	if err == nil {
+		return true
+	}
 	var configMap = corev1.ConfigMap{}
-	err := cli.Get(ctx, types.NamespacedName{Name: HACBSConfigMapName, Namespace: namespace}, &configMap)
+	err = cli.Get(ctx, types.NamespacedName{Name: HACBSConfigMapName, Namespace: namespace}, &configMap)
```
Comment (psturc) [inline gitops/prepare/prepare.go:123]: why do we need to check for hacbs configmap? does it mean that it'd be possible to trigger hacbs pipelines from appstudio workspace?

**Review by psturc — COMMENTED** [review]
just a question and a small suggestion

File: gitops/prepare/prepare.go (hunk 2 above)
Comment (Michkov) [reply gitops/prepare/prepare.go:123]: backward compatibility, APIBinding cannot be used for non-kcp usage.

File: gitops/prepare/prepare.go (hunk 1 above)
Comment (Michkov) [reply gitops/prepare/prepare.go]: updated

File: gitops/prepare/prepare.go (hunk 2 above)
Comment (tkdchen) [reply gitops/prepare/prepare.go:123]: Doesn't appstudio require kcp?

Comment (Michkov) [reply gitops/prepare/prepare.go:123]: Based on ASC-165 there should be still possible to run it on OpenShift.

File: main.go [hunk 3]
```diff
@@ -126,7 +127,7 @@ func main() {
 		// Local client is needed to access the workspace where operator deployment is
 		localClientScheme := runtime.NewScheme()
 		if err := clientgoscheme.AddToScheme(localClientScheme); err != nil {
-			setupLog.Error(err, "error adding standard types local client sceme")
+			setupLog.Error(err, "error adding standard types local client scheme")
```
Comment (mmorhun) [inline main.go:130]:
```suggestion
                        setupLog.Error(err, "error adding standard types to local client scheme")
```

**Review by mmorhun — APPROVED** [review]

**Review by johnmcollier — APPROVED** [review]

**Pull request merged**